### PR TITLE
Abhay/cuna

### PR DESCRIPTION
**Pull request by abhay009**
Added functionality to handle multiple pbix files for making connection to same dataflow.

### CODE REVIEW

File: Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Deployment.Common.Model;
+
+namespace Microsoft.Deployment.Actions.Common.PBI
+{
+    [Export(typeof(IAction))]
+    public class WranglePBICDSA : BaseAction
+    {
+        private const int RETRY_ATTEMPTS = 10;
+
+        private FileStream AVAwareOpen(string path, FileMode mode, FileAccess access, FileShare share)
```
Comment (cdoroftei) [inline Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs]: needed?

File: Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            if (result == null)
+                throw new AggregateException($"Could not open file: {path}", exceptionList);
+
+            return result;
+        }
+
+        public override async Task<ActionResponse> ExecuteActionAsync(ActionRequest request)
+        {
+            string connectionString = request.DataStore.GetValueAtIndex("SqlConnectionString", "SqlServerIndex");
```
Comment (cdoroftei) [inline Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs]: shouldn't this change the connection in the PBIX and make it point to CDSA and not SQL?

File: Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if (await this.MS.DeploymentService.executeActions() && !this.isUninstall) {
+            this.MS.DeploymentService.isFinished = true;
+            await this.wrangle();
+
+            if (this.pbixDownloadLinks.length > 1)
+                this.redirectInSameTab = false;
+
+            let index: number = 1;
+            for (let pbixDownloadLink of this.pbixDownloadLinks) {
+                let response: string = await this.MS.HttpService.getResponseAsync('Microsoft-PublishPBIReportCDSA', {
```
Comment (cdoroftei) [inline Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts]: If you're wrangling against CDSA you'll need the dataflowid too.

File: Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    async wrangle(): Promise<void> {
+        let response: ActionResponse = null;
+
+        response = await this.MS.HttpService.executeAsync('Microsoft-WranglePBICDSA', { FileName: this.filename });
+        
+        if (response.IsSuccess) {
+            this.pbixDownloadLinks = JSON.parse(response.Body.Value);
+        }
+
+        if (this.hasPowerApp) {
```
Comment (cdoroftei) [inline Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts]: if this is purely for CUNA you can remove this

File: Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs (hunk 1 above)
Comment (abhay009) [reply Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs]: Removed

File: Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs (hunk 2 above)
Comment (abhay009) [reply Source/Actions/Microsoft.Deployment.Actions.Common/PBI/WranglePBICDSA.cs]: Removed that part.

File: Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts (hunk 3 above)
Comment (abhay009) [reply Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts]: Dataflow id is not needed here, its set as a parameter.

File: Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts (hunk 4 above)
Comment (abhay009) [reply Source/SiteCommon/Web/pages-gallery/cdsa-progress.ts]: Removed.

**Pull request closed without merging**